egs/wsj/s5/steps/nnet3/multilingual/allocate_multilingual_examples.py

```diff
@@ -0,0 +1,295 @@
+#!/usr/bin/env python
+
+# Copyright 2017 Pegah Ghahremani
+#
+# Apache 2.0.
+
+""" This script generates egs.archive.scp and ranges.* used for multilingual setup.
+    Also this script generates outputs.*.ark and weight.*.ark, where each line
+    corresponds to language-id and weight for the same example in egs.*.scp.
+    weight.*.ark is archive of table indexed by the key of input example
```

Comment:
Is this definition fine with you?
    ranges.*.txt are files, each corresponding to one 'egs' file that will be   
    generated, where each line corresponds to ranges of  examples selected from 
    one of the input languages's scp files. where each line is:                 
    <lang> <local-scp-line> <num-examples>                                      
                                                                                
    That can be interpreted as selecting <num-example> examples starting from   
    <local-scp-line> line from {lang}_th 'egs' file in egs_scp_list.            
    (note that <local-scp-line> is the zero-based line number.)                 
                                                                                
    e.g. exp/tdnn_multilingual_sp/egs/temp/ranges.*.txt contains something like the following: 
    e.g.                                                                        
    0 0 256                                                                     
    2 1024 256